src/util/index.js

```diff
@@ -1,34 +1,39 @@
 // @flow
 
-const Mo = typeof MutationObserver === 'function' ? MutationObserver : class {
-  func: Function;
-  constructor (func) {
-    this.func = func;
-  }
-  observe (node: HTMLElement) {
-    const { func } = this;
-    const prop: Object = {
-      set () {
-        if (typeof Promise === 'undefined') {
-          setTimeout(func);
-        } else {
-          new Promise(resolve => resolve()).then(func);
+const Mo =
+  typeof MutationObserver === 'function'
+    ? MutationObserver
+    : class {
+        func: Function;
+        constructor(func) {
+          this.func = func;
         }
-      }
-    };
-    Object.defineProperty(node, 'textContent', prop);
-  }
-};
+        observe(node: HTMLElement) {
+          const { func } = this;
+          const prop: Object = {
+            set() {
+              if (typeof Promise === 'undefined') {
+                setTimeout(func);
+              } else {
+                new Promise(resolve => resolve()).then(func);
+              }
+            }
+          };
+          Object.defineProperty(node, 'textContent', prop);
+        }
+      };
 
-export function dashCase (str: string): string {
-  return str.split(/([_A-Z])/).reduce((one, two, idx) => {
-    const dash = !one || idx % 2 === 0 ? '' : '-';
-    two = two === '_' ? '' : two;
-    return `${one}${dash}${two.toLowerCase()}`;
-  });
+export function dashCase<T>(str: T): T {
```

Comment:
why generic? I guess it tightly set to be `string | symbol` right?

Reply:
Yeah, it is. So if I make it `string | Symbol`, then I get an error on the first and last lines [here](https://github.com/skatejs/skatejs/blob/3807c71ab54738d1703ded15bd04c21640b07a3c/src/util/with-unique.js#L20-L23) saying "Symbol This type is incompatible with the expected param type of string". Not really sure what to do here to fix that because `Ctor.name` is a string and we've typed it as such. Making it a generic at least made it have the same output as the input.